python/lsst/daf/butler/core/_butlerUri/http.py

```diff
@@ -501,5 +526,10 @@ def _emptyPut(self) -> requests.Response:
         response : `requests.Response`
             HTTP Response from the endpoint.
         """
+        if expect100():
```

Comment:
Rather than duplicating the entire call here I'd prefer it if we did it without any code duplication. Something like:

```python
headers = {"Content-Length": "0"}
if useExpect100():
    headers["Expect"] = "100-continue"
return self.session.put(self.geturl(), data=None, headers=headers,
                        allow_redirects=False, timeout=_defaultTimeout())
```